suricata/package.yaml

```diff
@@ -1,33 +1,1229 @@
+---
 id: suricata
 name: Suricata
 author: Tenzir
 author_icon: https://github.com/tenzir.png
-package_icon: https://suricata.io/wp-content/uploads/2023/09/Logo-Suricata-vert-R.png
-
+package_icon: |
+  https://suricata.io/wp-content/uploads/2023/09/Logo-Suricata-vert-R.png
 description: |
   A package to simplify the onboarding of logs emitted from a Suricata
   instance.
-
 inputs:
   eve-path:
     name: EVE Path
     description: A path where suricata writes eve.json formatted logs
     type: string
     default: /var/lib/suricata/eve.sock
-
+  eve-address:
+    name: EVE Address
+    description: A network address that suricata forwards formatted logs to
+    type: string
+    default: tcp://0.0.0.0:0000
 pipelines:
-  import:
-    name: Suricata Import
-    description: Ingest Suricata events into the internal storage of the node.
+  publish-from-file:
+    name: Suricata Publish
+    description: Ingest Suricata events from a fileinto a subscription topic.
+    restart-on-error: 5m
+    disabled: false
+    definition: |
+      // tql2
+      load_file "{{ inputs.eve-path }}", follow=true
+      read_suricata
+      publish "suricata"
+  publish-from-network:
+    name: Suricata Publish
+    description: |
+      Ingest Suricata events from a network source into a subscription topic.
+    restart-on-error: 5m
+    disabled: true
+    definition: |
+      // tql2
+      load_file "{{ inputs.eve-address }}"
+      read_suricata
+      buffer 16Mi, policy="drop"
+      publish "suricata"
+  suricata-flow-to-ocsf:
+    name: Suricata Flow to OCSF Network Activity
+    description: Maps Suricata Flow events to an OCSF Network Activity event.
+    restart-on-error: 5m
+    disabled: false
+    definition: |
+      // tql2
+      subscribe "suricata"
+      where @name == "suricata.flow"
```

Comment:
Since other Suricata events may also embed flows, it might be better to detect the presence of a top-level "flow" field of type record. To then avoid the duplicate results, we probably want to deduplicate at the end before publishing again (with some timeout to limit memory usage).

This also applies to all other pipelines below.

Reply:
> Since other Suricata events may also embed flows
Is that so? I thought every event type would be cleanly separated and we can use the `flow_id` to piece together information. Can we not do that here, too?

Reply:
No, that's how EVE works—it's extensible by certain subobjects. So an alert event will usually contain a flow, or a dns, or an smb subobject that it is related to.

Reply:
I don't understand what the resulting OCSF mapping would be then. To extend the`suricata.alert` example: The `alert` data would map to a[ Detection Finding](https://schema.ocsf.io/1.3.0/classes/detection_finding?extensions=) event, which offers no fields for traffic/flow data. The `flow` data itself would map to a [Network Activity](https://schema.ocsf.io/1.3.0/classes/network_activity?extensions=) event, which offers no fields for incident data. 

But both at the same time? What would the result be, if not a fork that publishes both Network Activity and Detection Finding events?

Reply:
Oh, I see, so basically another pipeline that subscribes to the suricata topic, tries to detect a `flow`, and then maps the event to a Network Activity. With deduplication added. That sounds easy enough.

Reply:
I think all you'd need to do is _replace_ the `where @name == "flow"` with `where flow != null` and you're done. Or, ideally we'd have a `where this.contains("flow")`, but that isn't implemented yet.

Reply:
I'm unsure about `where flow != null` because there are Suricata events that contain a `flow`, but entirely with `null` values.
I can use `where flow.start != null`, however.

Reply:
I think this is likely not going to happen anymore with @IyeOnline's precise parser changes. To me this sounds like an artifact of merging schemas.

Reply:
What test data set are you using for this?

Reply:
Fair enough.
But a quick pipeline check with `where flow != null` yielded no results on the UHH node, which is what I am using to prototype these pipelines.

Reply:
That will only work if you use TQL2, as `where` in TQL1 was only able to address leaf fields.

```
// tql2
export
where flow != null
```

Note that this will be _really_ slow for `export`, as there won't be any predicate pushdown for this, as that still uses TQL1 expressions, but it doesn't matter much for the use case at hand.

Reply:
But I am using TQL2. All my pipelines are using TQL2 since this month.

Even with `export` being slow for this use case, would you have expected "No results"?
![image](https://github.com/user-attachments/assets/81472f0a-0e8d-40f7-8728-f9ae1c785b4c)

![image](https://github.com/user-attachments/assets/869f5776-bcc2-4d82-b02e-1b9cd59f51eb)

Reply:
Probably a question for @jachris; for me that works in TQL2.

Reply:
We're capturing this in a follow-up issue for now.